src/modules/blockchain/implementation/web3-service.js

```diff
@@ -406,10 +416,16 @@ class Web3Service {
 
     async callContractFunction(contractInstance, functionName, args) {
         let result;
-        while (result === undefined) {
+        if (CACHED_FUNCTIONS.ParametersStorage.includes(functionName)) {
+            result = this.getCachedValue(functionName);
+        }
+        while (!result) {
             try {
                 // eslint-disable-next-line no-await-in-loop
                 result = await contractInstance[functionName](...args);
+                if (CACHED_FUNCTIONS.ParametersStorage.includes(functionName)) {
+                    this.cacheParameter(functionName, result);
+                }
             } catch (error) {
```

Comment:
Is there a reason why we don't call "handleError" in the catch block anymore ? This was previously used to switch RPCs in case the error was due to RPC being down.
If this is intended, then the while loop is not necessary and can be removed.

Reply:
We switched to a new provider implementation in the `ethers.js` that automatically hits different RPC after defined timeout, `while` loop probably can be removed as we're not retrying the call anyway (because of throwing new error)